components/edit_post_modal/edit_post_modal.tsx

```diff
@@ -4,65 +4,117 @@
 
 import React from 'react';
 import {Modal} from 'react-bootstrap';
-import {FormattedMessage, injectIntl} from 'react-intl';
-import PropTypes from 'prop-types';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
 import classNames from 'classnames';
 
 import {Constants, ModalIdentifiers} from 'utils/constants';
-import {splitMessageBasedOnCaretPosition, postMessageOnKeyPress} from 'utils/post_utils.jsx';
-
-import {intlShape} from 'utils/react_intl';
+import {
+    splitMessageBasedOnCaretPosition,
+    postMessageOnKeyPress,
+} from 'utils/post_utils.jsx';
 import * as Utils from 'utils/utils.jsx';
-import {getTable, formatMarkdownTableMessage, isGitHubCodeBlock, formatGithubCodePaste} from 'utils/paste';
+import {
+    getTable,
+    formatMarkdownTableMessage,
+    isGitHubCodeBlock,
+    formatGithubCodePaste,
+} from 'utils/paste';
 
 import DeletePostModal from 'components/delete_post_modal';
 import EmojiPickerOverlay from 'components/emoji_picker/emoji_picker_overlay.jsx';
 import EmojiIcon from 'components/widgets/icons/emoji_icon';
 import Textbox from 'components/textbox';
+import TextboxClass from 'components/textbox/textbox';
 import TextboxLinks from 'components/textbox/textbox_links';
+import TextboxLinksClass from 'components/textbox/textbox_links/textbox_links';
+import {CustomEmoji, Emoji, SystemEmoji} from 'mattermost-redux/types/emojis';
 
 const KeyCodes = Constants.KeyCodes;
 const TOP_OFFSET = 0;
 const RIGHT_OFFSET = 10;
 
-class EditPostModal extends React.PureComponent {
-    static propTypes = {
-        canEditPost: PropTypes.bool,
-        canDeletePost: PropTypes.bool,
-        codeBlockOnCtrlEnter: PropTypes.bool,
-        ctrlSend: PropTypes.bool,
-        config: PropTypes.object.isRequired,
-        intl: intlShape.isRequired,
-        maxPostSize: PropTypes.number.isRequired,
-        shouldShowPreview: PropTypes.bool.isRequired,
-        useChannelMentions: PropTypes.bool.isRequired,
-
-        editingPost: PropTypes.shape({
-            post: PropTypes.object,
-            postId: PropTypes.string,
-            refocusId: PropTypes.string,
-            commentCount: PropTypes.number,
-            show: PropTypes.bool.isRequired,
-            title: PropTypes.string,
-            isRHS: PropTypes.bool,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-            addMessageIntoHistory: PropTypes.func.isRequired,
-            editPost: PropTypes.func.isRequired,
-            hideEditPostModal: PropTypes.func.isRequired,
-            openModal: PropTypes.func.isRequired,
-            setShowPreview: PropTypes.func.isRequired,
-        }).isRequired,
-    }
-
-    constructor(props) {
+type OpenModal = {
+    ModalId: string;
+    dialogType: any;
```

Comment:
Same here, any suggestions on how to type this?

Reply:
Should be something like `typeof React.Component`, assuming all of the uses are derived from such, which they should be.